rs/boundary_node/rate_limits/canister/interface.did

```diff
@@ -0,0 +1,111 @@
+type Timestamp = nat64;         // Represents timestamp in nanoseconds since the epoch (1970-01-01)
+type RuleId = text;             // Unique identifier for each rule
+type SchemaVersion = nat64;     // Version of the schema for encoding/decoding the rules
+type IncidentId = text;         // Unique identifier for each incident
+
+
+// Input structure for defining a rule with mandatory fields within a config
+type InputRule = record {
+  incident_id: IncidentId; // Unique identifier for the incident
+  rule_raw: blob;          // Raw rule data (in binary format), expected to be a valid json object
+  description: text;       // Textual description of the rule
+};
+
+// Output structure for rules
+// Optional fields rule_raw and description may remain hidden while the rule is under confidentiality restrictions
+type OutputRule = record {
+  id: RuleId;                        // Unique identifier for the rule
+  rule_raw: opt blob;                // Raw rule data (in binary format), expected to be a valid json object, none if the rule is currently confidential
+  description: opt text;             // Textual description of the rule, none if the rule is currently confidential
+  disclosed_at: opt Timestamp;       // Timestamp when the rule was disclosed, none if the rule is still confidential
+};
+
+type OutputConfig = record {
+  rules: vec OutputRule;
+};
+
+// Response structure for returning the requested configuration and associated metadata
+type OutputConfigResponse = record {
+    version: Version;         // Version of the configuration
+    active_since: Timestamp;  // Time when this configuration was added (became active)
+    config: OutputConfig;     // Contains the list of rules
+};
+
+// Verbose details of an individual rule
+// Optional rule_raw and description fields are for restricted publicly viewing access
+type OutputRuleMetadata = record {
+  id: RuleId;                      // Unique identifier for the rule
+  rule_raw: opt blob;              // Raw rule data (binary format), expected to be a valid json object, none if the rule is currently confidential
+  description: opt text;           // Textual description of the rule, none if the rule is currently confidential
+  disclosed_at: opt Timestamp;     // Timestamp when the rule was disclosed, none if the rule is still confidential
+  added_in_version: Version;       // Version when the rule was added (became active)
+  removed_in_version: opt Version; // Version when the rule was deactivated (removed), none if the rule is still active
+};
+
+type GetRuleByIdResponse = variant {
+  Ok: OutputRuleMetadata;
+  Err: text;
+};
+
+type GetConfigResponse = variant {
+  Ok: OutputConfigResponse;
+  Err: text;
+};
+
+type DiscloseRulesResponse = variant {
+  Ok;
+  Err: text;
+};
+
+type OverwriteConfigResponse = variant {
+  Ok;
+  Err: text;
+};
+
+type DiscloseRulesByRuleIdsResponse = variant {
+  Ok;
+  Err: text;
+};
+
+type DiscloseRulesByIncidentIdResponse = variant {
```

Comment:
I see that you have a lot of
```
type XResponse = variant {
  Ok;
  Err: text;
};
```
maybe it'd make sense to consolidate those somehow?

Reply:
These types could be consolidate at some later point. But once consolidated the extensibility/flexibility is lost. Probably some `Ok` variant will contain messages, i haven't yet carefully thought through.

Reply:
Yes, but couldn't that just be a new Response type? I didn't mean you can only respond with this smaller type. E.g
```
type EmptyResponse = variant {
  Ok;
  Err: text;
};

type ListResponse = variant {
  Ok: vec RuleId,
  Err: text;
};

service: {
  disclose: (DiscloseArg) -> EmptyResponse;
  list: (ListArg) -> ListResponse;
};
```